backend/core/test/user/user.e2e-spec.ts

```diff
@@ -398,4 +400,59 @@ describe("Applications", () => {
     const token = await getUserAccessToken(app, newUser.email, password)
     expect(token).toBeDefined()
   })
+
+  it("should allow user to update user profile throguh PUT /userProfile/:id endpoint", async () => {
+    const userCreateDto: UserCreateDto = {
+      password: "Abcdef1!",
+      passwordConfirmation: "Abcdef1!",
+      email: "userProfile@b.com",
+      emailConfirmation: "userProfile@b.com",
+      firstName: "First",
+      middleName: "Mid",
+      lastName: "Last",
+      dob: new Date(),
+      language: Language.en,
+    }
+    const userCreateResponse = await supertest(app.getHttpServer())
+      .post(`/user/`)
+      .set("jurisdictionName", "Alameda")
+      .send(userCreateDto)
+      .expect(201)
+
+    const userService = await app.resolve<UserService>(UserService)
+    const user = await userService.findByEmail(userCreateDto.email)
+
+    await supertest(app.getHttpServer())
+      .put(`/user/confirm/`)
+      .send({ token: user.confirmationToken })
+      .expect(200)
+
+    const userAccessToken = await getUserAccessToken(
+      app,
+      userCreateDto.email,
+      userCreateDto.password
+    )
+
+    const userProfileUpdateDto: UserProfileUpdateDto = {
+      id: userCreateResponse.body.id,
+      createdAt: userCreateResponse.body.createdAt,
+      updatedAt: userCreateResponse.body.updatedAt,
+      jurisdictions: userCreateResponse.body.jurisdictions,
+      ...userCreateDto,
+      currentPassword: userCreateDto.password,
+      firstName: "NewFirstName",
+    }
+
+    await supertest(app.getHttpServer())
+      .put(`/userProfile/${userCreateResponse.body.id}`)
+      .send(userProfileUpdateDto)
+      .expect(401)
```

Comment:
There should probably be a separate test for ensuring that an unauthorized person can't access this. Can you add another test that generates two users, A and B, and ensure that A cannot update B's profile. And another with an admin updating a user's profile that isn't their own.